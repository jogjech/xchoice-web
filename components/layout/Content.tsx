import React, { useEffect } from "react";
import { FunctionComponent } from "react";
import { Layout } from "antd";
import { useAuth0 } from "@auth0/auth0-react";
import styles from "./Content.module.css";

interface Props {
  requiredAuth: boolean;
}

const Content: FunctionComponent<Props> = ({ requiredAuth, children }) => {
  const { loginWithRedirect, isAuthenticated, isLoading } = useAuth0();

  useEffect(() => {
    if (isLoading || isAuthenticated || !requiredAuth) {
      return;
    }
    (async (): Promise<void> => {
      await loginWithRedirect();
    })();
  }, [isLoading, isAuthenticated, loginWithRedirect]);

  return (
    <Layout.Content id={styles.siteContent}>
      <div id={styles.innerContent}>{children}</div>
    </Layout.Content>
  );
};

export default Content;
