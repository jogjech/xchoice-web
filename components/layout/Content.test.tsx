import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Content from "./Content";

describe("Page content", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Content></Content>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains innerContent div", () => {
    const innerContent = wrapper.find("div#innerContent");
    expect(innerContent.exists()).toBeTruthy();
  });
});
