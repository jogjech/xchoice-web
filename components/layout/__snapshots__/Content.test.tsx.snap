// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page content renders correctly 1`] = `
<Content
  id="siteContent"
>
  <div
    id="innerContent"
  />
</Content>
`;
